docs/architecture/adr-065-store-v2.md

```diff
@@ -81,73 +82,71 @@ less intrusive. Specifically, we propose to:
+supports this.
 
-However, in order for the SC layer to remain lightweight and not duplicate a
-majority of the data held in the SS layer, we encourage node operators to keep
-tight pruning strategies.
+Pruning (managing state bloat) is not considered a feature of store v2 or SC in
+the abstract, although it is probably required for an SC implementation to operate
+at scale.
 
 #### State Storage (SS)
 
-In the RMS, we will expose a *single* `KVStore` backed by the same physical
-database that backs the SC layer. This `KVStore` will be explicitly namespaced
-to avoid collisions and will act as the primary storage for (key, value) pairs.
-
-While we most likely will continue the use of `cosmos-db`, or some local interface,
-to allow for flexibility and iteration over preferred physical storage backends
-as research and benchmarking continues. However, we propose to hardcode the use
-of RocksDB as the primary physical storage backend.
-
-Since the SS layer will be implemented as a `KVStore`, it will support the
-following functionality:
-
-* Range queries
-* CRUD operations
-* Historical queries and versioning
-* Pruning
-
-The RMS will keep track of all buffered writes using a dedicated and internal
-`MemoryListener` for each `StoreKey`. For each block height, upon `Commit`, the
-SS layer will write all buffered (key, value) pairs under a [RocksDB user-defined timestamp](https://github.com/facebook/rocksdb/wiki/User-defined-Timestamp-%28Experimental%29) column
-family using the block height as the timestamp, which is an unsigned integer.
-This will allow a client to fetch (key, value) pairs at historical and current
-heights along with making iteration and range queries relatively performant as
-the timestamp is the key suffix.
-
-Note, we choose not to use a more general approach of allowing any embedded key/value
-database, such as LevelDB or PebbleDB, using height key-prefixed keys to
-effectively version state because most of these databases use variable length
-keys which would effectively make actions likes iteration and range queries less
-performant.
+The goal of SS is to provide a modular storage backend, i.e. multiple implementations,
+to facilitate storing versioned raw key/value pairs in a fast embedded database.
+The responsibility and functions of SS include the following:
+
+* Provided fast and efficient queries for versioned raw key/value pairs
+* Provide versioned CRUD operations
+* Provide versioned batching functionality
+* Provide versioned iteration (forward and reverse) functionality
+* Provide pruning functionality
+
+All of the functionality provided by an SS backend should work under a versioned
+scheme, i.e. a user should be able to get, store, and iterate over keys for the latest
+and historical versions efficiently and a store key, which is used for name-spacing
+purposes.
+
+We propose to have three defaulting SS backends for applications to choose from:
+
+* RocksDB (cgo-based)
+* PebbleDB
+* SQLite (cgo-based)
```

Comment:
we should probably add a small list of how and what features are used by the dbs. for sql lite are we using multiple tables or a single table, etc..